plugins/gfproxyd/rest.go

```diff
@@ -0,0 +1,142 @@
+package gfproxyd
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+func gfproxydEnableHandler(w http.ResponseWriter, r *http.Request) {
+	// Collect inputs from URL
+	p := mux.Vars(r)
+	volname := p["name"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	//validate volume name
+	volinfo, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if volume is started
+	if volinfo.State != volume.VolStarted {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrVolNotStarted.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Check if gfproxyd is already enabled
+	if volume.IsGfproxydEnabled(volinfo) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrGfproxydAlreadyEnabled.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Transaction which starts gfproxyd on all nodes.
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+
+	txn.Ctx.Set("volname", volname)
+
+	//Lock on Volume Name
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = volinfo.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "gfproxyd-start.Commit",
+			Nodes:  txn.Nodes,
+		},
+		unlock,
+	}
+
+	err = txn.Do()
+	if err != nil {
+		logger.WithFields(log.Fields{
+			"error":   err.Error(),
+			"volname": volname,
+		}).Error("failed to start gfproxy daemon")
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// Enable gfproxyd
+	volinfo.Options[volume.VkeyFeaturesGfproxyd] = "on"
```

Comment:
This is not persistent. since not saved in store, you can add "storeVolume" transaction step. Ref: https://github.com/gluster/glusterd2/blob/master/plugins/bitrot/rest.go#L155

Reply:
Done

Reply:
Need to be updated before `txt.Do` and make volinfo available to "vol-option.UpdateVolinfo" using Ctx.Set (Before `txn.Do`)

    volinfo.Options[volume.VkeyFeaturesGfproxyd] = "on"

    if err := txn.Ctx.Set("volinfo", volinfo); err != nil {
    		logger.WithError(err).Error("failed to set volinfo in transaction context")
    		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
    		return
    }

Reply:
And i should reset it to off, if it fails to do the transaction?.

Reply:
You can add rollback function later which can unset from volinfo.

Reply:
And i should reset it to off, if it fails to do the transaction?